dracut/30disk-uuid/disk-uuid.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+cmdline=( $(</proc/cmdline) )
+cmdline_arg() {
+    local name="$1" value="$2"
+    for arg in "${cmdline[@]}"; do
+        if [[ "${arg%%=*}" == "${name}" ]]; then
+            value="${arg#*=}"
+        fi
+    done
+    echo "${value}"
+}
+
+# First, check for the cmdline option in case it was set.
+RANDOMIZE_DISK_GUID=$(cmdline_arg flatcar.randomize_disk_guid)
+
+# But we are no longer detecting the non-randomized UUID in GRUB,
+# so set the base guid here if the cmdline was not set.
+BASE_GUID="00000000-0000-0000-0000-000000000001"
+if [[ -z "$RANDOMIZE_DISK_GUID" ]]; then
+    RANDOMIZE_DISK_GUID="${BASE_GUID}"
+fi
+
+# Check that the by-diskuuid path exists. If it doesn't, this script is
+# useless, so highlight that by exiting with a failure code
+DEVPATH="/dev/disk/by-diskuuid"
+if [[ ! -e ${DEVPATH} ]]; then
+    echo "${DEVPATH} does not exist. Bailing out"
+    exit 1
+fi
+
+# Verify the device actually exists before trying to touch it
+DEVICE="${DEVPATH}/${RANDOMIZE_DISK_GUID}"
+if [[ -e "${DEVICE}" ]]; then
+    /usr/bin/cgpt repair ${DEVICE} && \
+    /usr/sbin/sgdisk --disk-guid=R ${DEVICE} && \
+    /usr/bin/udevadm settle
```

Comment:
The service failed [on Azure](http://localhost:9091/job/os/job/kola/job/azure/109/console) with exit code 1. Logs are [here](http://localhost:9091/job/os/job/kola/job/azure/109/execution/node/4/ws/_kola_temp/azure-2020-07-31-1513-48767/get_cluster_semver/jenkins-azure-25ac007432/journal.txt/*view*/)
I don't see why but I suggest to make the script more robust and report errors. I guess the `udevadm settle` command failed due to some unrelated timeout? We can either try to remove the call or log the error and just continue.

Reply:
On all other platforms no test failed. I guess the Azure machine was very slow and also had some other problem: In the log udev is hanging for some time.

Reply:
It seems that this `udevadm settle` failure on Azure is nothing new and already happens some times. Here is an output of a Azure test failing on the **main** branch – we see that `sgdisk` was running (_The operation has completed successfully_) which means that the failing process was `udevadm settle`:
```
Jul 31 12:54:38.845227 systemd[1]: Starting Generate new UUID for disk GPT dev/disk/by-diskuuid/00000000-0000-0000-0000-000000000001...
Jul 31 12:54:38.870575 cgpt[405]: Primary Header is updated.
Jul 31 12:54:38.870575 cgpt[405]: Secondary Entries is updated.
Jul 31 12:54:38.870575 cgpt[405]: Secondary Header is updated.
Jul 31 12:54:38.891154 systemd[1]: Found device Virtual_Disk OEM.
Jul 31 12:54:38.921469 systemd[1]: Found device Virtual_Disk EFI-SYSTEM.
Jul 31 12:54:38.929559 kernel: random: crng init done
Jul 31 12:54:39.018151 systemd[1]: Found device Virtual_Disk ROOT.
Jul 31 12:54:39.021250 systemd[1]: Reached target Initrd Root Device.
Jul 31 12:54:39.044066 systemd[1]: Found device Virtual_Disk USR-A.
Jul 31 12:54:40.020420 sgdisk[407]: The operation has completed successfully.
Jul 31 12:54:40.024299 kernel:  sda: sda1 sda2 sda3 sda4 sda6 sda7 sda9
Jul 31 12:55:38.791337 systemd-udevd[257]: eth0: Worker [304] processing SEQNUM=1346 is taking a long time
Jul 31 12:56:38.102290 systemd[1]: Finished dracut initqueue hook.
Jul 31 12:56:38.105000 audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=dracut-initqueue comm="systemd" exe="/usr/lib64/systemd/systemd" hostname=? addr=? terminal=? res=success'
Jul 31 12:56:38.108621 systemd[1]: Reached target Remote File Systems (Pre).
Jul 31 12:56:38.128010 kernel: audit: type=1130 audit(1596200198.105:10): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=dracut-initqueue comm="systemd" exe="/usr/lib64/systemd/systemd" hostname=? addr=? terminal=? res=success'
Jul 31 12:56:38.121946 systemd[1]: Reached target Remote File Systems.
Jul 31 12:56:38.123074 systemd[1]: Starting dracut pre-mount hook...
Jul 31 12:56:38.147712 kernel: audit: type=1130 audit(1596200198.134:11): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=dracut-pre-mount comm="systemd" exe="/usr/lib64/systemd/systemd" hostname=? addr=? terminal=? res=success'
Jul 31 12:56:38.134000 audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=dracut-pre-mount comm="systemd" exe="/usr/lib64/systemd/systemd" hostname=? addr=? terminal=? res=success'
Jul 31 12:56:38.133674 systemd[1]: Finished dracut pre-mount hook.
Jul 31 12:56:40.162062 systemd[1]: disk-uuid@dev-disk-by\x2ddiskuuid-00000000\x2d0000\x2d0000\x2d0000\x2d000000000001.service: Main process exited, code=exited, status=1/FAILURE
Jul 31 12:56:40.197052 kernel: audit: type=1130 audit(1596200200.163:12): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=disk-uuid@dev-disk-by\x2ddiskuuid-00000000\x2d0000\x2d0000\x2d0000\x2d000000000001 comm="systemd" exe="/usr/lib64/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Jul 31 12:56:40.163000 audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=disk-uuid@dev-disk-by\x2ddiskuuid-00000000\x2d0000\x2d0000\x2d0000\x2d000000000001 comm="systemd" exe="/usr/lib64/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Jul 31 12:56:40.162240 systemd[1]: disk-uuid@dev-disk-by\x2ddiskuuid-00000000\x2d0000\x2d0000\x2d0000\x2d000000000001.service: Failed with result 'exit-code'.
Jul 31 12:56:40.162544 systemd[1]: Failed to start Generate new UUID for disk GPT dev/disk/by-diskuuid/00000000-0000-0000-0000-000000000001.
Jul 31 12:56:40.165358 systemd[1]: Dependency failed for Initrd Default Target.
Jul 31 12:56:40.166221 systemd[1]: initrd.target: Job initrd.target/start failed with result 'dependency'.
```
Full log [here](http://localhost:9091/job/os/job/kola/job/azure/108/execution/node/4/ws/_kola_temp/azure-2020-07-31-0952-38338/coreos.ignition.security.tls/jenkins-azure-d46235a78d/journal.txt/*view*/) for the test on the **main** branch (well almost, I tested a networkd unit to exclude weave devices but this is totally unrelated).

Reply:
Ok, I'm applying your changes. I'm not sure ignoring a udev failure is the way to go, though.